backend/server/acl.go

```diff
@@ -271,8 +299,42 @@ func aclMiddleware(s *Server, pathPrefix string, ce *casbin.Enforcer, next echo.
 				aclErr = enforceWorkspaceDeveloperProjectRouteACL(s.licenseService.GetEffectivePlan(), path, method, c.QueryParams(), principalID, roleFinder)
 			} else if strings.HasPrefix(path, "/sheet") {
 				aclErr = enforceWorkspaceDeveloperSheetRouteACL(s.licenseService.GetEffectivePlan(), path, method, principalID, roleFinder, sheetFinder)
-			}
+			} else if strings.HasPrefix(path, "/database") {
+				isMemberOfAnyProjectOwnsDatabase := func(principalID int, databaseID int) (bool, error) {
```

Comment:
Why we need to use callback here.

Seems like we can just define it inside enforceWorkspaceDeveloperDatabaseRouteACL

Reply:
Want to add unit test easily, but forgot commit...